### PR TITLE
use future-rs latest

### PR DESCRIPTION
**Pull request by conorbros**
Now that https://github.com/rust-lang/futures-rs/pull/2591 has been merged we can use the latest futures-rs version from git and remove the library code from our repo.

This PR is mostly just adding required traits and fixing some formatting.

### CODE REVIEW

**Comment by rukai** [comment]
Importing both futures git and futures_util 3.0 seems really wonky to me.
The reason is that futures reexports pats of futures_util.
This means we have traits from 2 versions of futures_util floating around which will cause a lot of confusion and hard to understand errors.

So I would rather we either wait for futures 4.0 or backport the push_front method to futures 3.0

File: shotover-proxy/Cargo.toml [hunk 1]
```diff
@@ -20,8 +20,9 @@ tokio-util = { version = "0.7.0", features = ["full"] }
 tokio-stream = "0.1.2"
 bytes = "1.0.0"
 bytes-utils = "0.1.1"
-futures = "0.3.12"
+futures = { git = "https://github.com/rust-lang/futures-rs" }
 futures-core = "0.3.1"
```
Comment (rukai) [inline shotover-proxy/Cargo.toml:24]: futures-core and pin-project-lite can be deleted now as they were only used by src/concurrency/mod.rs

**Comment by conorbros** [comment]
Marking as a draft until a new version of futures-rs is released.

File: shotover-proxy/Cargo.toml [hunk 2]
```diff
@@ -20,8 +20,8 @@ tokio-util = { version = "0.7.0", features = ["full"] }
 tokio-stream = "0.1.2"
 bytes = "1.0.0"
 bytes-utils = "0.1.1"
-futures = "0.3.12"
-futures-core = "0.3.1"
+futures = { git = "https://github.com/rust-lang/futures-rs" }
```
Comment (rukai) [inline shotover-proxy/Cargo.toml:23]: This can be changed back to crates.io now

Comment (conorbros) [reply shotover-proxy/Cargo.toml:23]: Yep, just fixing up the merge errors.

File: shotover-proxy/src/transforms/cassandra/connection.rs [hunk 3]
```diff
@@ -89,7 +90,7 @@ async fn tx_process<C: CodecWriteHalf, T: AsyncWrite>(
     codec: C,
 ) -> Result<()> {
     let in_w = FramedWrite::new(write, codec);
-    let rx_stream = UnboundedReceiverStream::new(out_rx).map(|x| {
+    let rx_stream = TokioStreamExt::map(UnboundedReceiverStream::new(out_rx), |x| {
```
Comment (rukai) [inline shotover-proxy/src/transforms/cassandra/connection.rs:93]: Are these `TokioStreamExt::map` changes still needed?
From memory these were due to issues with crate version mismatches?

Comment (conorbros) [reply shotover-proxy/src/transforms/cassandra/connection.rs:93]: I get a "multiple map found error"

Comment (rukai) [reply shotover-proxy/src/transforms/cassandra/connection.rs:93]: I reverted the entire file to origin/main and then reintroduced `use crate::concurrency::FuturesOrdered;` -> `use futures::stream::FuturesOrdered;` and it compiles fine.

Same with server.rs

Comment (conorbros) [reply shotover-proxy/src/transforms/cassandra/connection.rs:93]: Cool, I'll fix that up. Thanks

File: shotover-proxy/src/transforms/redis/sink_cluster.rs [hunk 4]
```diff
@@ -936,7 +936,7 @@ impl Transform for RedisSinkCluster {
         let mut responses = FuturesOrdered::new();
 
         for message in message_wrapper.messages {
-            responses.push(match self.dispatch_message(message).await {
+            responses.push_front(match self.dispatch_message(message).await {
```
Comment (rukai) [inline shotover-proxy/src/transforms/redis/sink_cluster.rs:939]: shouldnt this be push_back?

Comment (conorbros) [reply shotover-proxy/src/transforms/redis/sink_cluster.rs]: Yep, good catch. Fixed that up.

**Review by rukai — APPROVED** [review]
LGTM, just need to rebase

**Review by benbromhead — APPROVED** [review]

**Pull request merged**